File: python/1117.py
grade1 = 0
grade2 = 0

while True:
    grade1 = float(input())
    if grade1 >= 0 and grade1 <= 10:
        break
    else:
        print('nota invalida')
        continue
    
    
while True:
    grade2 = float(input())
    if grade2 >= 0 and grade2 <= 10:
        break
    else:
        print('nota invalida')
        continue
    
print(f'media = {((grade1 + grade2) / 2):.2f}')
    

File: lua/1117.lua
local sum = 0
local count = 0

while count < 2 do
    local grade = tonumber(io.read())
    if grade >= 0 and grade <= 10 then
        count = count + 1
        sum = sum + grade
    else
        print("nota invalida")
    end
    if count == 2 then
        print(string.format('media = %.2f', sum / 2))
    end
end